### PR TITLE
Update SonicTriton client to Triton version 2.3

### PR DESCRIPTION
**Pull request by kpedro88**
#### PR description:

Triton version 2 redesigns the client API. Most of the redesign is just reorganization and renaming. Important Triton changes:
1. client-side timeouts now supported
2. https/ssl now supported (not introduced here, support will come in a followup)
3. batch size handling changed: models that support triton-style batching have the specified shape modified so that the first entry is the batch size. The handling of shapes in SonicTriton is modified so that the user does not have to worry about this, and can continue to consider batch size and input shape separately, which is conceptually easier.

SonicTriton interface changes:
1. `modelVersion = cms.string()` (was `cms.int32()`)
2. `setShape()` accessors to change one dimension or all dimensions (rather than non-const ref `shape()` accessor)
3. unneeded `dims()` accessor removed
4. some other accessors made private (only used by `TritonClient`)

The cvmfs server image for tests is updated accordingly (see https://github.com/lgray/triton-torchgeo-gat-example/pull/1).

The corresponding external update is https://github.com/cms-sw/cmsdist/pull/6293.

#### PR validation:

Ran examples in `HeterogeneousCore/SonicTriton/test`.

### CODE REVIEW

**Comment by kpedro88** [comment]
@smuzaffar @mrodozov what is the right approach to make the code checks work with new headers from an external update?

**Comment by kpedro88** [comment]
@makortel I reran the code-checks and code-format following the instructions in https://github.com/cms-sw/cmsdist/pull/6293#issuecomment-706159297, and no changes were made.

**Comment by kpedro88** [comment]
@makortel any comments?

File: HeterogeneousCore/SonicTriton/src/TritonData.cc [hunk 1]
```diff
@@ -1,38 +1,104 @@
...
-  dname_ = ni::DataType_Name(dtype_);
-  //get byte size for input conversion
-  byteSize_ = ni::GetDataTypeByteSize(dtype_);
+TritonData<IO>::TritonData(const std::string& name, const TritonData<IO>::TensorMetadata& model_info, bool noBatch)
+    : name_(name),
+      dims_(model_info.shape().begin(), model_info.shape().end()),
+      noBatch_(noBatch),
+      batchSize_(0),
+      fullShape_(dims_),
+      shape_(&*(fullShape_.begin() + (noBatch_ ? 0 : 1)), &*(fullShape_.end())),
```
Comment (makortel) [inline HeterogeneousCore/SonicTriton/src/TritonData.cc:31]: De-referencing `std::vector<T>::end()` is undefined behavior.

If the type of `ShapeView` should always correspond to the type of `fullShape_`, how about
```cpp
using ShapeType = std::vector<int64_t>;
using ShapeView = edm::Span<ShapeType::const_iterator>;
...
ShapeType fullShape_;
```
and here
```suggestion
      shape_(fullShape_.begin() + (noBatch_ ? 0 : 1), fullShape_.end()),
```
? That would work even if the `vector::iterator` would be something else than a pointer.

**Comment by kpedro88** [comment]
@makortel thanks for the suggestion, my latest commit also includes code-checks and code-format

**Comment by makortel** [comment]
+1

**Comment by kpedro88** [comment]
@silviodonato @qliphy this will need to be force-merged (known problem: bot can't run code checks if CMSSW code uses new headers from an external PR)

please make sure to merge this together with https://github.com/cms-sw/cmsdist/pull/6297

thanks

**Comment by silviodonato** [comment]
merge

**Pull request merged**